backend/src/services/data_fetcher.py

```
@@ -194,31 +251,112 @@ async def get_all_tokens(self) -> List[Dict]:
+        if not BIRDEYE_API_KEY:
+            raise ValueError("BIRDEYE_API_KEY is required for token security analysis.")
+
+        try:
+            headers = {"X-API-KEY": BIRDEYE_API_KEY, "x-chain": "solana"}
+            url = f"{BIRDEYE_BASE_URL}/token_security?address={token_address}"
+            response = await self.http_client.get(url, headers=headers)
+            response.raise_for_status()
+            data = response.json()
+            if data.get('success'):
+                return data.get('data')
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)